src/install/lockfile.zig

```diff
@@ -2522,25 +2629,71 @@ pub const Package = extern struct {
                 out_group_i += 1;
             }
 
+            if (features.workspaces) {
+                out_groups[out_group_i] = DependencyGroup.workspaces;
+                out_group_i += 1;
+            }
+
             break :brk out_groups;
         };
 
+        var workspace_names: []string = undefined;
         inline for (dependency_groups) |group| {
             if (json.asProperty(group.prop)) |dependencies_q| {
-                if (dependencies_q.expr.data == .e_object) {
-                    for (dependencies_q.expr.data.e_object.properties.slice()) |item| {
-                        const key = item.key.?.asString(allocator) orelse "";
-                        const value = item.value.?.asString(allocator) orelse "";
+                switch (dependencies_q.expr.data) {
+                    .e_array => |arr| {
+                        workspace_names = try allocator.alloc(string, arr.items.len);
+
+                        var workspace_log = logger.Log.init(allocator);
+                        defer log.deinit();
+
+                        var workspace_buf = try allocator.alloc(u8, 1024);
```

Comment:
this can be a stack allocated buffer but with size 0 if the group is not workspaces

Reply:
Only `workspaces` have an array structure - all other dependency types are defined by object literals.

Reply:
Skip allocating altogether even when zero-sized `workspaces` is defined in https://github.com/oven-sh/bun/pull/1764/commits/c933a28c2ebb02eda1c342d6a6f346bea53c1c1c